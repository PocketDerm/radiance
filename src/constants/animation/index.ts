export default {
  defaultTiming: '350ms',
};
