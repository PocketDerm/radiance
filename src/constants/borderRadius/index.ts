const BORDER_RADIUS = {
  small: '4px',
  medium: '8px',
  large: '32px',
} as const;

export default BORDER_RADIUS;
