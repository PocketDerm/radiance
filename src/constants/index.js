export { default as ANIMATION } from './animation';
export { default as BOX_SHADOWS } from './boxShadows';
export { default as BREAKPOINTS } from './breakpoints';
export { default as COLORS } from './colors';
export { default as FONTS } from './fonts';
export { default as GRID } from './grid';
export { default as LAYOUT } from './layout';
export { default as MEDIA_QUERIES } from './mediaQueries';
export { default as PROGRESS_BAR_STATUS } from './progressBarStatus';
export { default as SPACING } from './spacing';
export { default as SPACER } from './spacer';
export { default as TYPOGRAPHY_CONSTANTS } from './typography';
export { default as Z_SCALE } from './zScale';
