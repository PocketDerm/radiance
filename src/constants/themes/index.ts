export { primaryTheme } from './primaryTheme';
export { secondaryTheme } from './secondaryTheme';
export * from './types';
