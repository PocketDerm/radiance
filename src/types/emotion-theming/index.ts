import type { ThemeType } from '../../constants/themes/types';
/**
 * We rely on TypeScript path aliasing in order to decorate our `emotion-theming` imports
 * with built-in type information, so for this one import we need to use relative imports
 *
 * @see: https://emotion.sh/docs/typescript#define-a-theme
 */
import {
  useTheme as useEmotionTheme,
  ThemeProvider,
} from '../../../node_modules/emotion-theming';

export const useTheme = () => useEmotionTheme<ThemeType>();

export { ThemeProvider };
