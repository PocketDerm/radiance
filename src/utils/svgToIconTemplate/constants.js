const UTIL_LOCATION = '../../../utils/icons';

module.exports = {
  UTIL_LOCATION,
};
