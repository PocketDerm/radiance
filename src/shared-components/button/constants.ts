import { COLORS } from '../../constants';

// TODO: potentially break out these pairings to the color constants file
export const textColorsAssociatedWithColors = {
  [COLORS.primary]: {
    tint1: COLORS.primaryTint1,
    tint2: COLORS.primaryTint2,
  },
  [COLORS.secondary]: {
    tint1: COLORS.lavender80,
    tint2: COLORS.lavender60,
  },
};
