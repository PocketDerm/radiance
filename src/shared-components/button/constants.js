import { COLORS } from 'src/constants';

//TODO: potentially break out these pairings to the color constants file
export const textColorsAssociatedWithColors = {
    purple: {
    	tint1: COLORS.purple85,
    	tint2: COLORS.purple70,
    },
    primary: {
    	tint1: COLORS.purple85,
    	tint2: COLORS.purple70,
    },
    purple100: {
    	tint1: COLORS.purple85,
    	tint2: COLORS.purple70,
    },
    lavender100: {
    	tint1: COLORS.lavender80,
    	tint2: COLORS.lavender60,
    },
    secondary: {
    	tint1: COLORS.lavender80,
    	tint2: COLORS.lavender60,
    },
};
