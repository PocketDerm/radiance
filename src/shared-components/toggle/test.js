import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { primaryTheme } from 'src/constants/themes';
import { ThemeProvider } from 'emotion-theming';

import Toggle from './index';

const ToggleWithTheme = (props) => (
  <ThemeProvider theme={primaryTheme}>
    <Toggle {...props} />
  </ThemeProvider>
);

describe('<Toggle />', () => {
  const labelText = 'Label Text';

  describe('UI snapshot', () => {
    it('renders the component', () => {
      const component = renderer.create(
        <ToggleWithTheme checked={false} label={labelText} />,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when label is undefined', () => {
    test('does not render a label component', () => {
      const wrapper = shallow(<ToggleWithTheme checked={false} />);
      expect(wrapper.html().indexOf('label') === -1).toBe(true);
    });
  });

  describe('when label is a string', () => {
    test('renders a text component', () => {
      const wrapper = shallow(
        <ToggleWithTheme checked={false} label={labelText} />,
      );

      expect(wrapper.html().indexOf(labelText) > 0).toBe(true);
    });
  });

  describe('when checkbox is clicked', () => {
    test('fires onChange function with correct argument when function exists', () => {
      const spy = jest.fn();
      const wrapper = mount(<ToggleWithTheme checked={false} onChange={spy} />);

      wrapper.find('[type="checkbox"]').simulate('click');
      expect(spy).toHaveBeenCalled();
    });
  });
});
