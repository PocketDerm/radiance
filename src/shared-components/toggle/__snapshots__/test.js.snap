// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> UI snapshot renders the component 1`] = `
.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 1rem 0;
}

.emotion-2 > input {
  display: none;
}

.emotion-0 {
  color: #524D6E;
  margin: 0;
  font-size: 1rem;
  line-height: 1.5rem;
  text-align: left;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin-right: 0.5rem;
}

<div
  className="emotion-2 emotion-3"
>
  <span
    className="emotion-0 emotion-1"
  >
    Label Text
  </span>
  <div
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "KhtmlUserSelect": "none",
        "MozUserSelect": "none",
        "WebkitTapHighlightColor": "transparent",
        "WebkitTouchCallout": "none",
        "WebkitUserSelect": "none",
        "alignItems": "center",
        "backgroundColor": "transparent",
        "border": 0,
        "cursor": "pointer",
        "display": "flex",
        "justifyContent": "flex-start",
        "msUserSelect": "none",
        "padding": 0,
        "position": "relative",
        "userSelect": "none",
        "width": 52,
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(237,237,240)",
          "borderRadius": 100,
          "display": "flex",
          "height": 24,
          "justifyContent": "center",
          "padding": 0,
          "width": 40,
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "color": "#FAFAFA",
            "display": "flex",
            "fontFamily": "'Helvetica Neue', Helvetica, sans-serif",
            "fontSize": 11,
            "height": 20,
            "justifyContent": "center",
            "left": 4,
            "lineHeight": 0,
            "marginBottom": "auto",
            "marginTop": "auto",
            "opacity": 0,
            "position": "relative",
            "width": 26,
          }
        }
      >
        
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "bottom": "0px",
            "color": "rgba(255,255,255,0.6)",
            "display": "flex",
            "fontFamily": "'Helvetica Neue', Helvetica, sans-serif",
            "fontSize": 11,
            "height": 20,
            "justifyContent": "center",
            "lineHeight": 0,
            "marginBottom": "auto",
            "marginTop": "auto",
            "opacity": 1,
            "paddingRight": 5,
            "position": "relative",
            "width": 26,
          }
        }
      >
        
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "display": "flex",
          "flex": 1,
          "height": "100%",
          "justifyContent": "flex-start",
          "left": 0,
          "position": "absolute",
          "top": 0,
        }
      }
    >
      <div
        style={
          Object {
            "MozBoxSizing": "border-box",
            "WebkitBoxSizing": "border-box",
            "alignSelf": "center",
            "background": "#ffffff",
            "backgroundColor": "rgb(250,250,250)",
            "border": "none",
            "borderRadius": "50%",
            "boxShadow": "none",
            "boxSizing": "border-box",
            "display": "flex",
            "height": 22,
            "left": 2,
            "position": "relative",
            "width": 22,
          }
        }
      />
    </div>
    <input
      onClick={[Function]}
      style={
        Object {
          "border": 0,
          "clip": "rect(0 0 0 0)",
          "height": 1,
          "margin": -1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "width": 1,
        }
      }
      type="checkbox"
      value={false}
    />
  </div>
</div>
`;
